SCSH_3C_iOS/SCSH_3C_iOS/Repository/BoardRepository.swift

```diff
@@ -7,55 +7,73 @@
 
 import Foundation
 import Combine
+import Persistence
+import Networking
 
 protocol BoardRepositoryType {
     func getExploreBoards(isLatest: Bool) -> AnyPublisher<[ExploreBoard], Error>
 }
 
 class BoardRepository: BoardRepositoryType {
     
-    // TODO: API design is required.
+    private let mapper: ExploreBoardMapperType
+    private let boardCoreDataService: BoardCoreDataServiceType
+    private let moyaNetworkFacade: MoyaNetworkFacadeType
+    
+    private let coreDataBackgroundContext = Persistence.CoreDataController.shared.container.newBackgroundContext()
+
+    init(boardCoreDataService: BoardCoreDataServiceType, moyaNetworkFacade: MoyaNetworkFacadeType, mapper: ExploreBoardMapperType) {
+        self.mapper = mapper
+        self.boardCoreDataService = boardCoreDataService
+        self.moyaNetworkFacade = moyaNetworkFacade
+    }
 
-    ///
-    /// ```
-    /// GET https://{domainhost}/boards
-    /// ?code={code}&limit={limit}`
-    /// ```
-    ///
-    /// Response: {
-    /// ```
-    ///     "data": [
-    ///         { "id": ...,
-    ///           "image_url": ...,
-    ///           "action": ...},
-    ///     ]
-    /// ```
-    /// }
-    ///
     func getExploreBoards(isLatest: Bool) -> AnyPublisher<[ExploreBoard], Error> {
-        Future { promise in
-            promise(.success([
-                ExploreBoard(id: "山跟海",
-                      imageUrl: URL(string: "https://images.pexels.com/photos/15484171/pexels-photo-15484171/free-photo-of-a-black-and-white-photo-of-the-ocean.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2"),
-                      action: URL(string: "https://images.pexels.com/photos/15484171/pexels-photo-15484171/free-photo-of-a-black-and-white-photo-of-the-ocean.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2")),
-                
-                ExploreBoard(id: "發光的車",
-                      imageUrl: URL(string: "https://images.pexels.com/photos/18395712/pexels-photo-18395712/free-photo-of-bmw-e30-parked-on-the-desert.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2"),
-                      action: nil),
+        if isLatest {
+            return fetchExploreBoards()
+                .flatMap { [weak self] _ -> AnyPublisher<[ExploreBoard], Error> in
+                    guard let self = self else {
+                        return Empty<[ExploreBoard], Error>().eraseToAnyPublisher()
+                    }
+                    
+                    return self.getExploreBoards()
+                }
+                .eraseToAnyPublisher()
+        } else {
+            return getExploreBoards()
+        }
```

Comment:
Maybe we can remove the else branch, if the main of `if else` branch ends with return.

```suggestion
        }
        return getExploreBoards()
```

Reply:
it's more concise, agree!!

Reply:
https://github.com/No-57/SCSH_3C_iOS/pull/45/commits/ff7742b99a4fbfec375f9884f445e893b57746da